airbyte-cdk/bulk/core/extract/src/main/kotlin/io/airbyte/cdk/read/StateManager.kt

```diff
@@ -37,18 +37,15 @@ class StateManager(
                     .mapKeys { it.key.id }
         } else {
             val globalStreams: Map<Stream, OpaqueStateValue?> =
-                global.streams.associateWith { initialStreamStates[it] }
+                global.streams.associateWith { initialStreamStates[it] } +
+                    initialStreamStates.filterKeys { global.streams.contains(it).not() }
```

Comment:
> contains(it).not()
I assume `contains()` is done by comparing reference.
is it guaranteed that initialStreamStates and global.streams contain the same objects?

Reply:
I don't think it compares by reference. It's not guaranteed that `initialStreamStates` and `global.streams` contain the same objects: for instance FULL_REFRESH streams can be in the former but not in the latter. INCREMENTAL streams can be in both.

Reply:
I just realized that `.contains()` should be replaced with the `!in` operator, for more idiomatic kotlin